### ISSUE TITLE
Configure LDAP authentication without user autocreation

### ISSUE THREAD

**Issue by Keisial** [MISP/MISP#1933]
It should be possible to configure LDAP just for authenticating existing users, without automatically creating new MISP accounts.

Seems it would be enough to have a config variable that is checked at line 97 of ApacheAuthenticate.php to return from authenticate(), skipping from line 98 onwards.

**Comment by truckydev** [comment]
I try to do this.
I managed to get the home page but nothing else works.
Because, you need to return a complet user object to do that. (user, role, perm, ...)

Many things will not work properly.(logs, restriction, see an event, ...) and more I don't think it's very safty to add this in prod.

regards

**Comment by MarSalfer** [comment]
Isn't user autocreation rather a security bug?
Automatically adding users from LDAP means compromising a MISP instance's confidentiality.

**Comment by RichieB2B** [comment]
No, you just hand off the authentication process to LDAP. So if you have a valid account (with the proper authorization group) in LDAP you get access to MISP. No security bug there to automatically create a MISP account when those criteria are met.
